apps/web/pages/api/user/avatar.ts

```diff
@@ -40,14 +45,18 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       shouldDefaultBeNameBased: true,
       avatar: team?.logo,
     };
+    linksToThisRoute =
+      identity.avatar === `${CAL_URL}/team/${teamname}/avatar.png` ||
+      identity.avatar === `${WEBAPP_URL}/team/${teamname}/avatar.png`;
   }
 
   const emailMd5 = crypto
     .createHash("md5")
     .update((identity?.email as string) || "guest@example.com")
     .digest("hex");
   const img = identity?.avatar;
-  if (!img) {
+  // If image isn't set or links to this route itself, use default avatar
+  if (!img || linksToThisRoute) {
```

Comment:
@hariombalhara Can we not write this as 
`
if( !linksToThisRoute)
`
because in the case where username/teamname are falsy values, linksToThisRoute will be false and also while setting the identity we are indirectly checking for img

Reply:
It has to execute for `linksToThisRoute` and not  `!linksToThisRoute`